docs/docs/Operators.md

```diff
@@ -0,0 +1,208 @@
+This document is a WIP proposal for the complete bytecode set supported by Base9. Today, Base9 only supports a limited
+few integer operations.
+
+# B9 VM's Instruction Set
+
+Base9 VM executes an untyped, stack-oriented, complex instruction set.
+
+## Operator Listing
+
+### Constants and literals
+```lisp
+(push-true)                 ;; ( -- true)
+(push-false)                ;; ( -- false )
+(push-null)                 ;; ( -- null )
+(push-undefined)            ;; ( -- undefined )
+
+(push-int-const integer)    ;; ( -- integer )
+(push-double-const double)  ;; ( -- double )
+(push-string id)            ;; ( -- string )
+(push-symbol id)            ;; ( -- symbol )
+(push-const-object id)      ;; ( -- object )
+```
+
+Constants are typed instructions. This allows us to abstract the encoding mechanism. Large constants, such as literal arrays or strings, are referenced as index
+### Stack Operations
+```lisp
+(duplicate)
+(drop n)
+```
+
+### Locals
+```lisp
+(push-from-local index)  ;; ( -- value )
+(pop-into-local  index)  ;; ( value -- )
+```
+
+### Arguments
+```lisp
+(push-from-argument index)  ;; ( -- value )
+(push-from-argument-i)      ;; ( index -- value )
+(pop-to-argument index)     ;; ( value -- )
+(push-arguments)            ;; ( -- arguments )
+(push-rest)                 ;; ( -- rest )
+```
+
+### Globals
+```lisp
+(push-from-global id)
+(pop-into-global  id)
+```
+
+The global object is a special 
+
+### Environment Register
+```lisp
+(push-from-environment hops index)  ;; ( -- value )
+(pop-into-environment  hops index)  ;; ( value -- )
+(enter-environment nslots)          ;; ( -- )
+(leave-environment)                 ;; ( -- )
+```
+
+The environment register is used to share state across function scopes. The environment is a chain 
+
+### This Register
+```lisp
+(call-with-this)  ;; ( args* callee this -- result )
+(push-this)       ;; ( )
```

Comment:
I assume you mean
`(push-this)       ;;( -- this)`

Reply:
derp, yes.